src/completion/tab_handler.rs

```diff
@@ -146,4 +154,22 @@ mod test {
         tab.handle(&mut new_buf);
         assert_eq!(new_buf, buffer_with("exit"));
     }
+
+    #[test]
+    fn same_string_different_places() {
+        let mut tab = get_tab_handler_with(vec!["that", "this"]);
+        let mut buf = buffer_with("th is my test th");
+
+        // Hitting tab after `th` fills the first completion `that`
+        buf.set_insertion_point(0, 2);
+        tab.handle(&mut buf);
+        let mut expected_buffer = buffer_with("that is my test th");
+        expected_buffer.set_insertion_point(0, 4);
+        assert_eq!(buf, expected_buffer);
+
+        // updating the cursor to end should reset the completions
+        buf.set_insertion_point(0, 18);
+        tab.handle(&mut buf);
+        assert_eq!(buf, buffer_with("that is my test that"));
+    }
```

Comment:
@jntrnr does this test capture the requirement you had in mind?

Reply:
Yup, exactly